### PR TITLE
Feature news banner

### PR DESCRIPTION
**Pull request by khihadysucahyo**
## Changes
- add news banner struct
- add news banner repo & usecase
- add news banner endpoint

## Payload:
Screen Shot 2021-11-01 at 10 23 45

@jabardigitalservice/jds-backend 

## Evidence:
project: Portal Jabar
title: Menembahkan endpoint untuk news banner
participants: @khihadysucahyo @rachadiannovansyah @firmanJS @ayocodingit @setiadijoe

### CODE REVIEW

File: core-service/src/modules/news/repository/mysql/mysql_news.go [hunk 1]
```diff
@@ -85,7 +84,7 @@ func (m *mysqlNewsRepository) Fetch(ctx context.Context, params *domain.Request)
 		query = query + ` AND title like '%` + params.Keyword + `%' `
```
Comment (setiadijoe) [inline core-service/src/modules/news/repository/mysql/mysql_news.go:83]: I think it should use query param such as `?` or named param to handle sql injection

File: core-service/src/modules/news/usecase/news_ucase.go [hunk 2]
```diff
@@ -80,6 +81,67 @@ func (n *newsUsecase) fillAuthorDetails(c context.Context, data []domain.News) (
 	return data, nil
 }
 
+func (n *newsUsecase) fillRelatedNews(c context.Context, data []domain.NewsBanner) ([]domain.NewsBanner, error) {
```
Comment (firmanJS) [inline core-service/src/modules/news/usecase/news_ucase.go:84]: Method `newsUsecase.fillRelatedNews` has 5 return statements (exceeds 4 allowed).

by the way codeclimate is off for this rule return ?

**Review by firmanJS — APPROVED** [review]

**Review by ayocodingit — APPROVED** [review]

**Review by rachadiannovansyah — APPROVED** [review]

File: core-service/src/modules/news/usecase/news_ucase.go (hunk 2 above)
Comment (khihadysucahyo) [reply core-service/src/modules/news/usecase/news_ucase.go:84]: 98bc010 already use `named return` to reduce exceeds allowed return

anyway thank for reminding us, as soon we'll pluge cc integration

File: core-service/src/modules/news/repository/mysql/mysql_news.go (hunk 1 above)
Comment (khihadysucahyo) [reply core-service/src/modules/news/repository/mysql/mysql_news.go:83]: actually we already prepare context for static query condotion e.g
```
query = query + ` LIMIT ?,? `
res, err = m.fetch(ctx, query, params.Offset, params.PerPage)
```

but not yet find practice if the query is optional e.g
`query = query + ` AND title like '%` + params.Keyword + `%' ``
the query above is only append when keyword is set.


any ideas @setiadijoe ? and can you give us example how SQL injection work on this case?

anyway thank for reminding us @setiadijoe  about SQL injection, we'll considering refactoring to the next PR when we understood the practice.

Comment (setiadijoe) [reply core-service/src/modules/news/repository/mysql/mysql_news.go:83]: I usually add parameters for LIKE in one variable and just use named parameter or `?` for in the query
```
var keyword = fmt.Sprintf('%%s%', params.Keyword)
query = query + AND title LIKE ?
```

**Pull request merged**